test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs

```diff
@@ -591,6 +591,33 @@ public void ItBuildsWhenReferencingWindowsUIXamlTypesWithUseUwpProperty()
                 .Pass();
         }
 
+        [WindowsOnlyFact]
+        public void ItHandlesProfilesWithSelfContained()
+        {
+            TestProject testProject = new()
+            {
+                TargetFrameworks = $"{ToolsetInfo.CurrentTargetFramework}-windows",
+                IsExe = true,
+                SelfContained = "true",
+                RuntimeIdentifier = "win-x64"
+            };
+            //  Setting both UseWpf and UseWindowsForms to true will add a FrameworkReference to Microsoft.WindowsDesktop.App
+            testProject.AdditionalProperties["UseWpf"] = "true";
+            testProject.AdditionalProperties["UseWindowsForms"] = "true";
+
+            //  Add reference to Windows Forms, which is a profile of Microsoft.WindowsDesktop.App
+            testProject.AddItem("FrameworkReference", "Include", "Microsoft.WindowsDesktop.App.WindowsForms");
```

Comment:
Is the reference to `Microsoft.WindowsDesktop.App.WindowsForms` required for this scenario to work correctly? A similar [published test scenario](https://github.com/dotnet/sdk/blob/8c3e741f90cc09ac7b17e07296787216f1234eae/test/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishTrimmedWindowsFormsAndWPFApps.cs#L256) seems to have the `PresentationFramework`.dll file.

Reply:
To repro the bug, the project needs a `FrameworkReference` to `Microsoft.WindowsDesktop.App` and to one (but not both) of the frameworks that use profiles: `Microsoft.WindowsDesktop.App.WPF` and `Microsoft.WindowsDesktop.App.WindowsForms`.  The SDK logic is that if both `UseWindowsForms` and `UseWPF` are set to true, then it will add a reference to `Microsoft.WindowsDesktop.App` instead of the individual profile references.  That's what happens in the test you pointed to.

The real-world scenario where this is likely to happen is with a project that has `UseWindowsForms` and `UseWPF` set to true that references a NuGet package that depends on one of the profile frameworks.